### ISSUE TITLE
Tiktok account username change

### ISSUE THREAD

**Issue by jean75000** [roju/tiktok-live-recorder#33]
Hello, 

I use the script automatically and sometimes the script no longer finds information on a user who has changed the name of their Tiktok account. Is it not possible to add a check that detects the change and modifies the -user variable with the new Tiktok account name ?

Thanks

**Comment by jean75000** [comment]
OK Thanks

**Comment by jean75000** [comment]
@Marksmanship256  

I don't want to open another ticket for this and I saw that you have a good command of python, I am a beginner, I think I have a false positive (probably linked to my VPN) of blacklist and it stops the script, when I restart the script it starts again although I have not done anything.

ERROR - Captcha required or country blocked. Use a vpn or room_id.
To get room id: https://github.com/Michele0303/TikTok-Live-Recorder/blob/main/GUIDE.md#how-to-get-room_id
Unrestricted country list: https://github.com/Michele0303/TikTok-Live-Recorder/edit/main/GUIDE.md#unrestricted-country
2024-07-22 22:45:12,068 - ERROR - Exception caught in main:
2024-07-22 22:45:12,068 - ERROR - Redirect

Traceback (most recent call last):
  File "/home/headless/tiktok_bot.py", line 291, in get_room_id_from_user
    raise errors.Blacklisted('Redirect')
errors.Blacklisted: Redirect

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/headless/ttlr.py", line 110, in main
    bot.run()
  File "/home/headless/tiktok_bot.py", line 98, in run
    raise e
  File "/home/headless/tiktok_bot.py", line 57, in run
    self.room_id = self.get_room_id_from_user()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/headless/tiktok_bot.py", line 304, in get_room_id_from_user
    raise errors.Blacklisted(e)

Basically I would like that when I have the error that it continues the script without stopping

**Comment by jean75000** [comment]
yes, I have already updated the script with the addition of cookies it worked perfectly again thanks :), but today I decided to put a VPN and suddenly I encounter the error randomly on users.

I modified that I don't know if it will correct my problem:

Before:

![image](https://github.com/user-attachments/assets/44564bc4-06dc-4381-b498-2e85371580b2)

After:

![image](https://github.com/user-attachments/assets/8ee9db84-9d0c-4783-aee8-b9ae183e5e78)


I had also modified the script on my side to add the functionality of cookies with selenium but I prefer your technique because it is less resource-intensive.

**Comment by jean75000** [comment]
after making the code modification that I indicated to you I did not have the error again if I have it again I will test your code in any case thank you very much for your help :)

**Comment by jean75000** [comment]
after several days/weeks I have no more problems, the script would be perfect if it could manage the change of account name I found how to do it but I don't know how to implement it in python can someone help me?

basically each user has a unique ID for their account it would be necessary when we put the parameter "python ttlr.py -user test" that it retrieves the id and that it is based only on this id because I did a test if we replace https://www.tiktok.com/@uidtest' it redirects well to the page of the user in question: https://www.tiktok.com/@test

to receive the uid you have to look in the source code of the user's page this: {"user": {"id":

**Comment by jean75000** [comment]
thanks but unfortunately it will not solve the problem if the user changes his account name the script will indicate that it cannot find the user :(

The most reliable way is to base it on the unique id because it does not change it is composed of numbers see screenshot

that's what interests me because I have users who change the name of their account frequently and I am annoyed

![image](https://github.com/user-attachments/assets/1e94e479-5ce1-4cf6-ac92-4a01bd892df2)
![image](https://github.com/user-attachments/assets/b453488f-9a22-4896-a76e-2715f7bca4f0)

**Comment by jean75000** [comment]
thanks but unfortunately it will not solve the problem if the user changes his account name the script will indicate that it cannot find the user :(

The most reliable way is to base it on the unique id because it does not change it is composed of numbers see screenshot

that's what interests me because I have users who change the name of their account frequently and I am annoyed

![image](https://github.com/user-attachments/assets/60cc0953-0dc7-43f0-835c-dadf3254e211)
![image](https://github.com/user-attachments/assets/facdfeb7-bad8-4025-af80-53c2798d0ac4)

https://github.com/user-attachments/assets/5a27a28c-481c-4992-9b0a-7e4efddc8a48

**Comment by jean75000** [comment]
I just changed my account name to test and my unique ID is exactly the same it has not changed so it is much more reliable to go through the unique ID than through the username

**Comment by jean75000** [comment]
Now we need to find a way to make it so that when we give it a username it finds the unique ID on its own and bases itself on it.

**Comment by jean75000** [comment]
basically you have to base yourself on the unique id because it is the only reliable element but to find it is a bit tedious and I have a list of usernames if I have to find the unique Id of each one I have it for the day :)

ideally we indicate the argument -user test
it checks the username and uid of the user
it is based on the uid to verify if the user is online
we should add a detection to indicate to us that the user has changed his username for example when we have the log the user is offline add the old and the new account of the user

what do you think?

**Comment by jean75000** [comment]
thanks for the help but I'm a bit lost I've never done this with chatgpt and I don't have a paid account.

and if I had to use the user id I wanted to do it on everyone

**Comment by broklin48** [comment]
> This works with both username and user ID.
> 
> ![test](https://private-user-images.githubusercontent.com/76043670/355511523-b4a350ec-c8fb-4441-8007-7efadc6ff450.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2NDQwODksIm5iZiI6MTcyMzY0Mzc4OSwicGF0aCI6Ii83NjA0MzY3MC8zNTU1MTE1MjMtYjRhMzUwZWMtYzhmYi00NDQxLTgwMDctN2VmYWRjNmZmNDUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDEzNTYyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzNjc3YmU1MjNhMWRlOTYzNTUzMTA1MDhjMDI2Y2M4ZjJiYjM5ZDYyN2UxOWRiMmI1NzA0Zjc1NTVkMDRjNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XuzIQCNrMTi4XDCfUCOxu_lHvqc3SIgwLZoq1jrmNug)
> 
> `https://www.tiktok.com/@6555704086321184769`

not working with some user id.
e.g. https://www.tiktok.com/@67361217230
